doc/modules/clustering.rst

```diff
@@ -880,6 +887,107 @@ by black points below.
+HDBSCAN can be seen as an algorithm which performs DBSCAN* clustering across all
+values of `eps`. As mentioned prior, this is equivalent to finding the connected
+components of the mutual reachability graphs for all values of `eps`. To do this
+efficiently, HDBSCAN first extracts a minimum spanning tree (MST) from the fully
+-connected mutual reachability graph, then greedily cuts the edges with heighest
+weight. An outline of the HDBSCAN algorithm is as follows:
+
+  1. Extract the MST of :math:`G_{ms}`
+  2. Extend the MST by adding a "self edge" for each vertex, with weight equal
+     to the core distance of the underlying sample.
+  3. Initialize a single cluster and label for the MST.
+  4. Remove the edge with the greatest weight is from the MST (ties are
+     removed simultaneously).
+  5. Assign cluster labels to the connected components which contain the
+     end points of the now-removed edge. If the component does not have at least
+     one edge it is instead assigned a "null" label marking it as noise.
+  6. Repeat 4-6 until there are no more connected components.
+
+.. note::
+
+  The clustering generated by taking the connected components of a trimmed
+  graph :math:`G_{ms,\epsilon}` equivalent to running DBSCAN* with `min_samples`
+  and `eps`. DBSCAN* is a slightly modified version of DBSCAN mentioned in
+  https://doi.org/10.1007/978-3-642-37456-2_14
+
+HDBSCAN is therefore able to obtain all possible partitions obtainable by
+DBSCAN* for a fixed choice of `min_samples` in a hierarchical fashion.
+Indeed, this allows HDBSCAN to perform clustering across multiple densities
+and as such it no longer needs `eps` to be given as a hyperparameter. Instead
+it relies solely on the choice of `min_samples`, which tends to be a more robust
+hyperparameter.
+
+.. |hdbscan_ground_truth| image:: ../auto_examples/cluster/images/sphx_glr_plot_hdbscan_001.png
+        :target: ../auto_examples/cluster/plot_hdbscan.html
+        :scale: 75
+.. |hdbscan_results| image:: ../auto_examples/cluster/images/sphx_glr_plot_hdbscan_006.png
+        :target: ../auto_examples/cluster/plot_hdbscan.html
+        :scale: 75
+
+.. centered:: |hdbscan_ground_truth|
+.. centered:: |hdbscan_results|
+
+HDBSCAN can be smoothed with an additional hyperparameter `min_cluster_size`
+which specifies that during the hierarchical clustering, components with fewer
+than `minimum_cluster_size` many samples are considered noise. In practice, one
+can set `minimum_cluster_size = min_samples` to couple the parameters and
+simplify the hyperparameter space.
+
+.. topic:: References:
+
+ * Campello, R.J.G.B., Moulavi, D., Sander, J. (2013). Density-Based Clustering
+   Based on Hierarchical Density Estimates. In: Pei, J., Tseng, V.S., Cao, L.,
+   Motoda, H., Xu, G. (eds) Advances in Knowledge Discovery and Data Mining.
+   PAKDD 2013. Lecture Notes in Computer Science(), vol 7819. Springer, Berlin,
+   Heidelberg. https://doi.org/10.1007/978-3-642-37456-2_14
```

Comment:
There is a `:doi:` role to use. For example:

https://github.com/scikit-learn/scikit-learn/blob/67a3feed2fe4e82c1cc129c34b9e223b94a8d531/doc/modules/kernel_approximation.rst?plain=1#L249-L251